src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -508,6 +550,7 @@ public double eventElapsedTime(final String eventName) {
      * See also {@link #track(String, org.json.JSONObject)}
      */
     public void trackMap(String eventName, Map<String, Object> properties) {
+        if (hasOptedOutTracking()) return;
```

Comment:
it's fine but you don't need this (`track(String eventName, JSONObject properties)` will do it)

Reply:
yeah - defensive programming since i can't guarantee this method won't change in the future